core/src/com/unciv/logic/civilization/transients/CivInfoTransientCache.kt

```diff
@@ -215,23 +217,63 @@ class CivInfoTransientCache(val civInfo: Civilization) {
             civInfo.addNotification("We have discovered [${tile.naturalWonder}]!",
                 tile.position, NotificationCategory.General, "StatIcons/Happiness")
 
-            var goldGained = 0
+            val statsGained = Stats()
+
             val discoveredNaturalWonders = civInfo.gameInfo.civilizations.filter { it != civInfo && it.isMajorCiv() }
                     .flatMap { it.naturalWonders }
-            if (tile.terrainHasUnique(UniqueType.GrantsGoldToFirstToDiscover)
+            if (tile.terrainHasUnique(UniqueType.GrantsStatsToFirstToDiscover)
                     && !discoveredNaturalWonders.contains(tile.naturalWonder!!)) {
-                goldGained += 500
+
+                for (unique in tile.getTerrainMatchingUniques(UniqueType.GrantsStatsToFirstToDiscover)) {
+                    statsGained.add(unique.stats)
+                }
             }
 
-            if (civInfo.hasUnique(UniqueType.GoldWhenDiscoveringNaturalWonder)) {
-                goldGained += if (discoveredNaturalWonders.contains(tile.naturalWonder!!)) 100 else 500
+            for (unique in civInfo.getMatchingUniques(UniqueType.StatBonusWhenDiscoveringNaturalWonder)) {
+
+                val normalBonus = Stats.parse(unique.params[0])
+                val firstDiscoveredBonus = Stats.parse(unique.params[1])
+
+                if (discoveredNaturalWonders.contains(tile.naturalWonder!!))
+                    statsGained.add(normalBonus)
+                else
+                    statsGained.add(firstDiscoveredBonus)
+            }
+
+            // Variable for support of twooo deprecated uniques
+            var goldGained = 0
+
+            // Support for depreciated GoldWhenDiscoveringNaturalWonder unique
+            for (unique in civInfo.getMatchingUniques(UniqueType.GoldWhenDiscoveringNaturalWonder)) {
+
+                goldGained += if (discoveredNaturalWonders.contains(tile.naturalWonder!!)) {
+                    100
+                } else {
+                    500
+                }
+            }
+
+            // Support for depreciated GrantsGoldToFirstToDiscover unique
+            if (tile.terrainHasUnique(UniqueType.GrantsGoldToFirstToDiscover)
+                && !discoveredNaturalWonders.contains(tile.naturalWonder!!)) {
+
+                for (unique in tile.getTerrainMatchingUniques(UniqueType.GoldWhenDiscoveringNaturalWonder)) {
+                    goldGained += 500
+                }
+            }
+
+
+            if (!statsGained.isEmpty()) {
+                civInfo.addStats(statsGained)
+                civInfo.addNotification("We have received [${statsGained}] for discovering [${tile.naturalWonder}]",
```

Comment:
I see that this new notification text does not yet exist.
You need to add it to template.properties, preferably right under "We have received [goldAmount] Gold for discovering [naturalWonder]"

Format should be:
```
We have received [statsGained] for discovering [naturalWonder] = 
```

Reply:
Since it's going to be deprecated in a month or so anyway, I don't really see a point. Allow current mods to keep using existing translations.

Reply:
Yes, keep both
We'll delete the old ones when the old unique is actually deprecated